hax/hax/server.py

```diff
@@ -281,6 +281,82 @@ def fn():
     return _process
 
 
+def event_subscribe_fn(data, consul_util):
+    resource_types = ['node']
+    interface_types = ['health_message']
+
+    for item in data.keys():
+        if item not in resource_types:
+            raise Exception(f'Invalid reource type({item})')
+        if data[item] not in interface_types:
+            raise Exception(f'Invalid interface type({data[item]})')
+
+        subscriber = consul_util.kv.kv_get(f'events/subscription/{item}',
+                                           allow_null=True)
+        subscriber_list = []
+        if subscriber:
+            subscriber_list = json.loads(subscriber['Value'])
+        if data[item] not in subscriber_list:
+            subscriber_list.append(data[item])
+        consul_util.kv.kv_put(f'events/subscription/{item}',
+                              json.dumps(subscriber_list))
+
+
+def event_unsubscribe_fn(data, consul_util):
```

Comment:
- please create a ha/utils.py and move this function there
- have a generic function, `event_subscribe()`
- have a specific function `node_event_subscribe()` that invokes `event_subscribe()` with node event to HealthMessage subscription.